File: src/main/kotlin/com/ks/culinario/network/controller/AuthController.kt
package com.ks.culinario.network.controller

import LoginRequestDTO
import com.ks.culinario.domain.service.AuthService
import com.ks.culinario.network.dto.AuthResponseDTO
import com.ks.culinario.network.dto.RefreshTokenRequest
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/auth")
class AuthController(private val authService: AuthService) {

    @PostMapping("/login")
    fun login(@RequestBody request: LoginRequestDTO): AuthResponseDTO {
        return authService.login(request)
    }

    @PostMapping("/refresh")
    fun refresh(@RequestBody request: RefreshTokenRequest): AuthResponseDTO {
        return authService.refresh(request)
    }

    @PostMapping("/logout")
    fun logout(authentication: Authentication) {
        val username = authentication.name
        authService.logout(username)
    }
}

File: dotnet/Culinario.API/Controllers/AuthController.cs
using Culinario.Application.DTOs;
using Culinario.Application.Services;
using Microsoft.AspNetCore.Mvc;

namespace Culinario.API.Controllers;

[ApiController]
[Route("api/[controller]")]
public class AuthController : ControllerBase
{
    private readonly IAuthService _authService;

    public AuthController(IAuthService authService)
    {
        _authService = authService;
    }

    [HttpPost("login")]
    public async Task<ActionResult<AuthResponseDTO>> Login(LoginRequestDTO request)
    {
        try
        {
            return await _authService.LoginAsync(request);
        }
        catch (Exception ex)
        {
            return Unauthorized(ex.Message);
        }
    }

    [HttpPost("refresh")]
    public async Task<ActionResult<AuthResponseDTO>> Refresh(RefreshTokenRequest request)
    {
        return await _authService.RefreshAsync(request);
    }

    [HttpPost("logout")]
    public async Task<IActionResult> Logout()
    {

        var username = User.Identity?.Name;
        if (string.IsNullOrEmpty(username))
        {
            return Unauthorized();
        }
        await _authService.LogoutAsync(username);
        return Ok();
    }
}
